### ISSUE TITLE
[ENH] test for two `fit` calls with different argument

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1613]
The current test `test_fit_idempotent` did not catch the bug in https://github.com/alan-turing-institute/sktime/pull/1595 because the test checks two calls to `fit` with the same arguments.
This fails to catch the case where, for instance, the resulting estimator behaves like after the first, but not the *second* `fit` call.

A better test would call `fit` twice, but with different arguments.
The resulting estimator should behave like the estimator fitted only by the *second* `fit` call.

This should be a new test, or a change to `test_fit_idempotent`.

**Comment by Carlosbogo** [comment]
Hi, I would be glad to help with this. Is there anyone already working on it?

**Comment by fkiraly** [comment]
I don't think so, please feel free to go ahead! You should be able to use the `_make_args` helper functions to get this done easily.

@ltsaprounis, FYI
